daffodil-core/src/test/scala/org/apache/daffodil/general/TestPrimitives.scala

```diff
@@ -137,6 +137,131 @@ class TestPrimitives {
+            <xs:element name="s5" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s6" type="xs:string" dfdl:lengthKind="delimited"/>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>)
+    val (_, actual) = TestUtils.testString(sch, "SingleSpace, MultSpace,   SingleTab,\u0009MultTab,\u0009\u0009Mix,\u0009\u0009 \u0009     \u0009End")
+    val expected: Node = <e1><s1>SingleSpace</s1><s2>MultSpace</s2><s3>SingleTab</s3><s4>MultTab</s4><s5>Mix</s5><s6>End</s6></e1>
+    TestUtils.assertEqualsXMLElements(expected, actual)
+  }
+
+  @Test def testLSPStarDelimiter(): Unit = {
+    val sch = SchemaUtils.dfdlTestSchema(
+      <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
+
+      <dfdl:format ref="tns:GeneralFormat" representation="text" lengthUnits="bytes" encoding="US-ASCII" initiator="" separator="" terminator="" ignoreCase="no"/>,
+
+      <xs:element name="e1" dfdl:lengthKind="delimited">
+        <xs:complexType>
+          <xs:sequence dfdl:separator=",%LSP*;" dfdl:separatorPosition="infix">
+            <xs:element name="s0" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s1" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s2" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s3" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s4" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s5" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s6" type="xs:string" dfdl:lengthKind="delimited"/>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>)
+    val (_, actual) = TestUtils.testString(sch, "None,SingleSpace, MultSpace,   SingleTab,\u0009MultTab,\u0009\u0009Mix,\u0009\u0009 \u0009     \u0009End")
+    val expected: Node = <e1><s0>None</s0><s1>SingleSpace</s1><s2>MultSpace</s2><s3>SingleTab</s3><s4>MultTab</s4><s5>Mix</s5><s6>End</s6></e1>
+    TestUtils.assertEqualsXMLElements(expected, actual)
+  }
+
+  @Test def testSPDelimiter(): Unit = {
+    val sch = SchemaUtils.dfdlTestSchema(
+      <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
+
+      <dfdl:format ref="tns:GeneralFormat" representation="text" lengthUnits="bytes" encoding="US-ASCII" initiator="" separator="" terminator="" ignoreCase="no"/>,
+
+      <xs:element name="e1" dfdl:lengthKind="delimited">
+        <xs:complexType>
+          <xs:sequence dfdl:separator=",%SP;" dfdl:separatorPosition="infix">
+            <xs:element name="s1" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s2" type="xs:string" dfdl:lengthKind="delimited"/>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>)
+    val (_, actual) = TestUtils.testString(sch, "SingleSpace, End")
```

Comment:
You need a test showing that if there are two spaces after the comma, that the s2 element contains the 2nd space. `<s2> End</s2>`